index.js

```diff
@@ -1,47 +1,73 @@
 const PrettyStream = require('bunyan-prettystream');
 
 const SensitiveDataStream = require('./streams/sensitive-data');
 
-const loggerLevel = process.env.LOGGER_LEVEL || 'info';
-const loggerName = process.env.LOGGER_NAME || 'Development logger';
-
-const config = {
-  name: loggerName,
-  level: loggerLevel,
-  streams: [],
-  serializers: {
-    err: bunyan.stdSerializers.err,
+const defaultConfig = {
+  logger: {
+    name: process.env.LOGGER_NAME || 'Development logger',
+    level: process.env.LOGGER_LEVEL || 'info',
+    pretty: process.env.USE_BUNYAN_PRETTY_STREAM === 'true',
+    hideSensitiveData: process.env.LOGGER_USE_SENSITIVE_DATA_STREAM !== 'false',
+    sensitiveDataPattern: process.env.LOGGER_SENSITIVE_DATA_PATTERN,
+  },
+  sentry: {
+    dsn: process.env.SENTRY_DSN,
+    release: undefined,
+    environment: process.env.NODE_ENV || 'development',
   },
 };
 
-if (process.env.USE_BUNYAN_PRETTY_STREAM === 'true') {
-  const prettyStdOut = new PrettyStream();
-  prettyStdOut.pipe(process.stdout);
-  config.streams.push({ type: 'raw', level: loggerLevel, stream: prettyStdOut });
-} else if (process.env.LOGGER_USE_SENSITIVE_DATA_STREAM !== 'false') {
-  config.streams.push({
+function init(config) {
+  const finalConfig = _.defaultsDeep(config, defaultConfig);
+
+  const loggerName = finalConfig.logger.name;
+  const loggerLevel = finalConfig.logger.level;
+  const loggerConfig = {
+    name: loggerName,
     level: loggerLevel,
-    stream: new SensitiveDataStream(process.env.LOGGER_SENSITIVE_DATA_PATTERN),
-  });
-} else {
-  config.streams.push({ level: loggerLevel, stream: process.stdout });
-}
+    streams: [],
+    serializers: {
+      err: bunyan.stdSerializers.err,
+    },
+  };
+
+  if (finalConfig.logger.pretty) {
+    const prettyStdOut = new PrettyStream();
+    prettyStdOut.pipe(process.stdout);
+    loggerConfig.streams.push({
+      type: 'raw',
+      level: loggerLevel,
+      stream: prettyStdOut,
+    });
+  } else if (finalConfig.logger.hideSensitiveData) {
+    loggerConfig.streams.push({
+      level: loggerLevel,
+      stream: new SensitiveDataStream(finalConfig.logger.sensitiveDataPattern),
+    });
+  } else {
+    loggerConfig.streams.push({ level: loggerLevel, stream: process.stdout });
+  }
+
+  if (finalConfig.sentry && finalConfig.sentry.dsn) {
+    const client = new raven.Client(finalConfig.sentry.dsn, {
+      name: loggerName,
+      release: finalConfig.sentry.release,
+      environment: finalConfig.sentry.environment,
+    });
+    client.install();
+    loggerConfig.streams.push(sentryStream(client));
+  }
 
-let client;
-if (process.env.SENTRY_DSN) {
-  client = new raven.Client(process.env.SENTRY_DSN, { name: loggerName });
-  client.install();
-  config.streams.push(sentryStream(client));
+  return bunyan.createLogger(loggerConfig);
 }
 
-// Logger instance:
-const logger = bunyan.createLogger(config);
+const logger = init(defaultConfig);
 
 module.exports = logger;
```

Comment:
As it's a breaking change, maybe we should force projects to only use the init method and not initialize the logger internally?

```js
module.exports = {
  init
} 
```

Reply:
Let's do it progressively until we are sure we will use the sentry releases everywhere